### PR TITLE
Observation Service Interface

### PR DESCRIPTION
**Pull request by terryyylim**
This PR contains the initial version for Observation Service which includes the following generated artifacts:

- Go gRPC , gRPC-gateway, and Open API client codes
- Python gRPC and Open API client codes
- Protobuf Documentation
- Open API Spec
NOTE: all files under gen folder can be ignored.

### CODE REVIEW

File: proto/caraml/upi/v1/observation_log.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package caraml.upi.v1;
+
+import "caraml/upi/v1/variable.proto";
+import "caraml/upi/v1/table.proto";
```
Comment (krithika369) [inline proto/caraml/upi/v1/observation_log.proto:6]: This import is not needed here, right?

**Review by krithika369 — COMMENTED** [review]
Are a;; the docs and `gen/` contents added by running the `make` command as described in the README: https://github.com/caraml-dev/universal-prediction-interface ?

**Comment by terryyylim** [comment]
>Are all the docs and gen/ contents added by running the make command as described in the README: https://github.com/caraml-dev/universal-prediction-interface ?

Yes it was all from `make generate`.

File: proto/caraml/upi/v1/observation_log.proto (hunk 1 above)
Comment (terryyylim) [reply proto/caraml/upi/v1/observation_log.proto:6]: We need it for `message Value`.

Comment (pradithya) [reply proto/caraml/upi/v1/observation_log.proto:6]: I think it's better to use `Variable` for the observation_values field, the `Value` type is tightly coupled with `Table` data type which store its type information.

**Review by pradithya — APPROVED** [review]
LGTM, thanks @terryyylim  . FYI @peterjrichens

**Pull request merged**